compiler/src/main/java/org/qbicc/type/definition/classfile/MethodParser.java

```diff
@@ -636,15 +636,17 @@ void processNewBlock() {
                     case OP_AALOAD: {
                         v2 = pop1();
                         v1 = pop1();
-                        v1 = gf.load(gf.elementOf(gf.referenceHandle(v1), v2), MemoryAtomicityMode.UNORDERED);
+                        ValueHandle handle = gf.referenceHandle(v1);
+                        v1 = gf.load(gf.elementOf(handle, v2), handle.getDetectedMode());
```

Comment:
Without this patch, if I try to access array of structs `opt` was crashing with following message:
```
atomic load operand must have integer, pointer, or floating point type!
```

Looking at the IR, I realized `load` node for loading an element of native array handle was getting lowered to `atomic load` instruction in LLVM IR because we are passing `MemoryAtomicityMode.UNORDERED` to the load node.

```
%T.struct.org.qbicc.runtime.methoddata.MethodData$ip_source_code_info = type { i64, i32 }
...
  %L11 = getelementptr [0 x %T.struct.org.qbicc.runtime.methoddata.MethodData$ip_source_code_info], [0 x %T.struct.org.qbicc.runtime.methoddata.MethodData$ip_source_code_info]* @qbicc_instruction_map_table, i32 0, i32 %p0, !dbg !16
  %L12 = load atomic %T.struct.org.qbicc.runtime.methoddata.MethodData$ip_source_code_info, %T.struct.org.qbicc.runtime.methoddata.MethodData$ip_source_code_info* %L11 unordered, align 8, !dbg !16 ; MethodData.java:83 bci@53
```

[Language ref](https://llvm.org/docs/LangRef.html#load-instruction) for `load` instruction states that for atomic load "type of the pointee must be an integer, pointer, or floating-point type".
This is the reason for changing the memory ordering to the handle's memory ordering instead of hard coding it to `UNORDERED`.

There are other places in `MethodParser` where the ordering for `load` and `store` is obtained from the value handle. eg: https://github.com/qbicc/qbicc/blob/d2245017cbb16bf926fc02903bd011e2f4d36d10/compiler/src/main/java/org/qbicc/type/definition/classfile/MethodParser.java#L1322
https://github.com/qbicc/qbicc/blob/d2245017cbb16bf926fc02903bd011e2f4d36d10/compiler/src/main/java/org/qbicc/type/definition/classfile/MethodParser.java#L1333